strawberry_django/ordering.py

```diff
@@ -51,17 +74,44 @@ def generate_order_args(order: WithStrawberryObjectDefinition, prefix: str = "")
         elif ordering == Ordering.DESC:
             args.append(f"-{prefix}{f.name}")
         else:
-            subargs = generate_order_args(ordering, prefix=f"{prefix}{f.name}__")
+            subargs = generate_order_args(
+                ordering,
+                prefix=f"{prefix}{f.name}__",
+                sequence=(seq := sequence.get(f.name)) and seq.children,
+            )
             args.extend(subargs)
 
     return args
 
 
-def apply(order: Optional[WithStrawberryObjectDefinition], queryset: _QS) -> _QS:
+def apply(
+    order: Optional[WithStrawberryObjectDefinition],
+    queryset: _QS,
+    info: Optional[Info] = None,
+) -> _QS:
     if order in (None, strawberry.UNSET):  # noqa: PLR6201
         return queryset
 
-    args = generate_order_args(order)
+    sequence: Dict[str, _OrderSequence] = {}
+    if info is not None and info._raw_info.field_nodes:
+        field_node = info._raw_info.field_nodes[0]
+        for arg in field_node.arguments:
+            if arg.name.value != ORDER_ARG or not isinstance(
+                arg.value, ObjectValueNode
+            ):
+                continue
+
+            def parse_and_fill(field: ObjectValueNode, seq: Dict[str, _OrderSequence]):
```

Comment:
@bellini666 
This function definition does not depend on any value of the for-loop. Maybe define it just before the for-loop?

Reply:
In this case the for is just to find the order argument, so this code should happen only once

Reply:
Yes I understand, but the `def parse_and_fill` is defined for every iteration of the `for`. It should probably defined just before the for-loop or you prefer to be defined in the for-loop for scope reason? So that `parse_and_fill` only exist within the for-loop?

Reply:
TBH I didn't think too much about that, the only reason I created the function was because I had a recursive code in there =P

Because I created it just to be used exactly in that next line, I decided to create it as close as possible.

Also, as I said, although the function in theory is defined for every iteration, in reality it isn't because all other iterations will `continue` in the previous `if`, only one will actually hit this code